src/ripple/ledger/ApplyView.h

```diff
@@ -371,6 +371,19 @@ class ApplyView
         return dirRemove (directory, page, key.key, keepRoot);
     }
     /** @} */
+
+    /** Remove the specified directory, if it is empty.
+
+        @param directory the identifier of the directory node to be deleted
+        @return \c true if the directory was found was succesfully deleted
+                \c false otherwise.
+
+        @note The function should only be called with the root entry (i.e. with
+              the first page) of a directory. Calling it with an intermediate
+              page can result in undefined behavior.
+    */
+    bool
+    dirDelete(Keylet const& directory);
```

Comment:
So `dirRemove` removes an entry (and possibly the directory as well) and `dirDelete` removes the directory. I think those two names are too similar to eachother. How about `rmDirEntry` and `rmDir`?

As a side note: both implementation are similar. We could probably refactor them to use a helper function, tho we may want to punt on both of these for this PR and do a refactor in another PR if we want.